swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/IPCClient.swift

```diff
@@ -97,6 +97,69 @@ enum IPCClient {
     try await sendMessageWithoutResponse(session: session, message: message)
   }
 
+  // MARK: - Log operations with utun cleanup
+
+  /// Wraps an IPC call with wake-up and cleanup logic when VPN is not connected.
+  ///
+  /// On macOS, IPC calls can wake the network extension, which creates a utun device.
+  /// If not properly cleaned up (via stop), these devices accumulate.
+  @MainActor
+  private static func wrapIPCCallIfNeeded<T>(
+    store: Store,
+    session: NETunnelProviderSession,
+    operation: @MainActor () async throws -> T
+  ) async throws -> T {
+    #if os(macOS)
+      if store.vpnStatus == .connected {
+        // Extension already running, utun already created, just execute
+        return try await operation()
+      } else {
```

Comment:
You already have an early return above so you can get rid of this `else` and save on rightwards-drift for the remaining logic.

Reply:
I do prefer visually to know the both alternatives being indented, and myself find the right drift helpful...
Happy to change it if you find it unbearable though.